clients/src/main/java/org/apache/kafka/clients/consumer/internals/DefaultBackgroundThread.java

```diff
@@ -168,28 +171,41 @@ public DefaultBackgroundThread(final Time time,
                             logContext);
             CoordinatorRequestManager coordinatorRequestManager = null;
             CommitRequestManager commitRequestManager = null;
+            HeartbeatRequestManager heartbeatRequestManager = null;
+            MembershipManager membershipManaber = null;
 
+            // TODO: consolidate groupState and memberState
             if (groupState.groupId != null) {
                 coordinatorRequestManager = new CoordinatorRequestManager(
-                        this.time,
-                        logContext,
-                        retryBackoffMs,
-                        retryBackoffMaxMs,
-                        this.errorEventHandler,
-                        groupState.groupId);
+                    this.time,
+                    logContext,
+                    retryBackoffMs,
+                    retryBackoffMaxMs,
+                    this.errorEventHandler,
+                    groupState.groupId);
```

Comment:
We should submit a patch to combine groupState into memberState. @lianetm

Reply:
Totally, but is there a reason why we couldn't use the MemberState here already? I though that was the point of unblocking the state PR (to be able to use it here)

Reply:
actually - my bad: I think we can just get rid of the groupState.

Reply:
There's a bit of refactoring needed, to keep things in scope for this PR - I will be submitting another PR to address this.